restapi/restapi.go

```diff
@@ -292,3 +339,111 @@ func (r *RestAPI) getSalesforceOauthToken() (string, error) {
 
 	return authResponse.AccessToken, nil
 }
+
+func (r *RestAPI) setDefaults() {
+	if r.ListMethod == "" {
+		r.ListMethod = r.Method
+	}
+	if r.ListMethod == "" {
+		r.ListMethod = http.MethodGet
+	}
+	if r.CreateMethod == "" {
+		r.CreateMethod = http.MethodPost
+	}
+	if r.BatchSize <= 0 {
+		r.BatchSize = DefaultBatchSize
+	}
+	if r.BatchDelaySeconds <= 0 {
+		r.BatchDelaySeconds = DefaultBatchDelaySeconds
+	}
+	if r.UserAgent == "" {
+		r.UserAgent = "personnel-sync"
+	}
+}
+
+func (r *RestAPI) addContact(p psync.Person, n *uint64, wg *sync.WaitGroup, eventLog chan<- psync.EventLogItem) {
+	defer wg.Done()
+
+	apiURL := fmt.Sprintf("%s%s", r.BaseURL, r.setConfig.CreatePath)
+	headers := map[string]string{"Content-Type": "application/json"}
+	responseBody, err := r.httpRequest(r.CreateMethod, apiURL, attributesToJSON(p.Attributes), headers)
+	if err != nil {
+		eventLog <- psync.EventLogItem{
+			Event: "error",
+			Message: fmt.Sprintf("addContact %s httpRequest error %s, response: %s", p.CompareValue, err,
+				responseBody),
+		}
+		return
+	}
+
+	eventLog <- psync.EventLogItem{
+		Event:   "AddContact",
+		Message: p.CompareValue,
+	}
+
+	atomic.AddUint64(n, 1)
+}
+
+func attributesToJSON(attr map[string]string) string {
+	jsonObj := gabs.New()
+	for field, value := range attr {
+		if _, err := jsonObj.SetP(value, field); err != nil {
+			log.Printf("error setting field %s for REST API add, %s", field, err)
+		}
+	}
+	return jsonObj.String()
+}
+
+func (r *RestAPI) updateContact(p psync.Person, n *uint64, wg *sync.WaitGroup, eventLog chan<- psync.EventLogItem) {
+	wg.Done()
+}
+
+func (r *RestAPI) deleteContact(p psync.Person, n *uint64, wg *sync.WaitGroup, eventLog chan<- psync.EventLogItem) {
+	wg.Done()
+}
+
+func (r *RestAPI) httpRequest(verb, url, body string, headers map[string]string) (string,
+	error) {
```

Comment:
Does this need to be split out to a new line like this?

Reply:
Nope, thanks.